src/pages/komodo-defi-framework/api/v20/enable_eth_with_tokens/index.mdx

```diff
@@ -13,11 +13,14 @@ Additionally, it supports ERC20 tokens on the ETH chain and associated ERC20 lik
-| fallback\_swap\_contract    | string           | Address of backup etomic swap smart contract                                                                                                                                                                                                                                                                                                 |
-| maker\_swap\_v2\_contract   | string           | Backup address for the maker's swap smart contract                                                                                                                                                                                                                                                                                           |
-| taker\_swap\_v2\_contract   | string           | Backup address for the taker's swap smart contract                                                                                                                                                                                                                                                                                           |
-| nodes                       | array of objects | A list of standard [EvmNode](/komodo-defi-framework/api/common_structures/activation/#evm-node) objects. ([https://github.com/KomodoPlatform/coins/tree/master/electrums](https://github.com/KomodoPlatform/coins/tree/master/electrums)).                                                                                                   |
-|                             |                  |                                                                                                                                                                                                                                                                                                                                              |
+
+| fallback\_swap\_contract    | string           | Address of backup etomic swap smart contract.                                                                                                                                                                                                                                                                                                |
+| maker\_swap\_v2\_contract   | string           | Address for the maker's new V2 swap smart contract. (Must be provided if "use_trading_proto_v2"is true in mm2 configuration)
+                                                                                                                                                                                               |
+| taker\_swap\_v2\_contract   | string           | Address for the taker's new V2 swap smart contract.  (Must be provided if "use_trading_proto_v2"is true in mm2 configuration)
```

Comment:
maker and taker swap v2 fields are not standalone, they belong to `swap_v2_contracts` field. Its optional and has currently three swap contract addresses.
```rust
swap_v2_contracts: Option<SwapV2Contracts>,
```
yep, if mm conf has "use_trading_proto_v2" flag true then  "swap_v2_contracts" must be provided. In other way error `"swap_v2_contracts must be provided when using trading protocol v2"` occurs.
In addition, addresses must be non zero, otherwise error `"All swap_v2_contracts addresses must be non-zero"` 

```curl --url "http://127.0.0.1:7783" --data '{
  "userpass": "'$USERPASS'",
  "method": "enable_eth_with_tokens",
  "mmrpc": "2.0",
  "params": {
    "ticker": "MATIC",
    "mm2": 1,
    "swap_contract_address": "0x9130b257D37A52E52F21054c4DA3450c72f595CE",
    "swap_v2_contracts":{
	    "maker_swap_v2_contract": "0x9130b257D37A52E52F21054c4DA3450c72f595CE",
	    "taker_swap_v2_contract": "0x9130b257D37A52E52F21054c4DA3450c72f595CE",
	    "nft_maker_swap_v2_contract": "0x9130b257D37A52E52F21054c4DA3450c72f595CE"
    },    
    "fallback_swap_contract": "0x9130b257D37A52E52F21054c4DA3450c72f595CE",
    "nodes": [
      {
        "url": "https://polygon-rpc.com"
      }
    ],
    "erc20_tokens_requests": []
  }
}'
```

Also in this text `if "use_trading_proto_v2"is true` space is missing.

PS: please note that swap_v2 contract addresses above are not for production. We haven’t deployed the new contracts yet.